client/jetpack-connect/authorize.js

```diff
@@ -99,8 +99,10 @@ export class JetpackAuthorize extends Component {
 
 	componentWillMount() {
 		const { recordTracksEvent } = this.props;
-		recordTracksEvent( 'calypso_jpc_authorize_form_view' );
-		recordTracksEvent( 'calypso_jpc_auth_view' );
+
+		const tracksProperties = pick( this.props.authQuery, 'from' );
+		recordTracksEvent( 'calypso_jpc_authorize_form_view', tracksProperties );
+		recordTracksEvent( 'calypso_jpc_auth_view', tracksProperties );
```

Comment:
Because all connections go through this event, we should be able to segment any other JPC event by `from` by creating a temp funnel with `calypso_jpc_auth_view` + the desired event.